orc8r/cloud/go/obsidian/access/rest_middleware.go

```diff
@@ -125,15 +127,27 @@ func TokenMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 			return echo.NewHTTPError(http.StatusBadRequest, "failed to parse basic auth header")
 		}
 
-		getPDReq := &certprotos.GetPolicyDecisionRequest{
+		resourceType, resourceVal := getResource(c)
+		getPDReq := certprotos.GetPolicyDecisionRequest{
 			Username: username,
 			Token:    token,
-			Resource: &certprotos.Resource{
+			Request: &certprotos.Request{
 				Action:   getRequestAction(req, nil),
 				Resource: req.RequestURI,
 			},
 		}
-		pd, err := certifier.GetPolicyDecision(req.Context(), getPDReq)
+		switch resourceType {
+		case constants.NetworkID:
+			getPDReq.Request.ResourceId = &certprotos.Request_NetworkId{NetworkId: resourceVal}
+		case constants.TenantID:
+			id, err := strconv.ParseInt(resourceVal, 10, 64)
+			if err != nil {
+				obsidian.MakeHTTPError(err, http.StatusInternalServerError)
+			}
```

Comment:
This is probably a 400 [0] since the client is passing a bad argument

[0] https://devdocs.io/http/status/400